packages/app/src/pages/transaction/index.tsx

```diff
@@ -0,0 +1,330 @@
+import React, { useState, useEffect } from 'react';
```

Comment:
I took a stab at refactoring and cleaning some things up. I'm a bit hesitant to go full redux here, yet, because it's only one page. It could benefit from a vanilla Reducer to clean up the state stuff, agreed. It's a bit better now, but still not beautiful.